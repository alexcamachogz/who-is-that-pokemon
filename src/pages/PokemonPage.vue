<template>
  <h1>¿Quién es este pokémon?</h1>
  <PokemonPicture :pokemon-id="251" :show-pokemon="true"/>
  <PokemonOptions :pokemons="pokeArray"/>
</template>

<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'
import getPokemonOptions from '@/helpers/getPokemonOptions'

export default {
  components: {
    PokemonOptions,
    PokemonPicture
  },

  data () {
    return {
      pokeArray: []
    }
  },

  methods: {
    async mixedPokeArray () {
      this.pokeArray = await getPokemonOptions()
    }
  },

  mounted () {
    this.mixedPokeArray()
  }
}
</script>
