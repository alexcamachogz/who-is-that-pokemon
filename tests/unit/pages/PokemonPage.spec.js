import { shallowMount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from '../mocks/pokemons.mock'

describe('PokemonPage Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage)
  })

  test('Should be match with the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Should be call mixedPokeArray when is mounted', () => {
    const mixedPokeArraySpy = jest.spyOn(PokemonPage.methods, 'mixedPokeArray')
    shallowMount(PokemonPage)
    expect(mixedPokeArraySpy).toHaveBeenCalled()
  })

  test('Should be match with the snapshot when pokemon load', () => {
    const wrapper = shallowMount(PokemonPage, {
      data () {
        return {
          pokeArray: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: ''
        }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
