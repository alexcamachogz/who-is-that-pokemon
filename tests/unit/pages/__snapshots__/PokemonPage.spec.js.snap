// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component Should be match with the snapshot 1`] = `<h1>Cargando...</h1>`;

exports[`PokemonPage Component Should be match with the snapshot when pokemon loads 1`] = `
<div class="game-container">
  <h1>¿Quién es este pokémon?</h1>
  <div class="game">
    <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub>
    <pokemon-options-stub pokemonlist="[object Object],[object Object],[object Object],[object Object]" election="false"></pokemon-options-stub>
    <h3>Puntaje: 0</h3>
  </div>
  <!--v-if-->
</div>
`;
